test/Amazon.Extensions.CognitoAuthentication.IntegrationTests/CognitoAWSCredentialsTests.cs

```diff
@@ -139,30 +139,30 @@ await identityClient.SetIdentityPoolRolesAsync(new SetIdentityPoolRolesRequest()
         /// Internal method that cleans up the created identity pool (along with associated 
         /// clients/roles) for testing
         /// </summary>
-        public override async void Dispose()
+        public override void Dispose()
         {
             try
             {
-                await identityClient.DeleteIdentityPoolAsync(new DeleteIdentityPoolRequest()
+                identityClient.DeleteIdentityPoolAsync(new DeleteIdentityPoolRequest()
                 {
                     IdentityPoolId = identityPoolId
-                }).ConfigureAwait(false);
+                }).Wait();
```

Comment:
Prefer .GetAwaiter().GetResult() over .Wait() and .Result.

For background check https://www.codurance.com/publications/2018/12/06/wait-and-getawaiter

```suggestion
                }).GetAwaiter().GetResult();
```

Reply:
Done